hatchways/team-gingerbread
# Integration of Profile Details

### What this PR does (required):
- addresses #62 
- renders ProfileDetails dynamically via URL (e.g. /profile-details/61affa655479b0ee3cc36331 shows profile details about the user with an id of 61affa655479b0ee3cc36331)
- BookingsForm is populated with the availability of the user of interest
- submitting a request creates a notification for that user that a service request has been placed

### Screenshots / Videos (front-end only):
![image](https://user-images.githubusercontent.com/78042694/145135033-c8bede90-47e7-4dce-bcba-4b15573c08c2.png)

### Any information needed to test this feature (required):
- run server and app
- login with your account
- copy and paste your user id into client/mocks/mockId.tsx
- navigate to /profile-details/<your-user-id-here>
- the profile details card should populate with the information in your profile (can change via EditProfile)
- make an appointment via BookingsForm (make sure your profile has availableTime property in database; the code that changes the profile model to include this property is in FE/IN Availability Page for reference)
![image](https://user-images.githubusercontent.com/78042694/145134941-4421a603-2344-4d1f-bbd3-f911cf4f21c5.png)
- check mongoDB to see if new notification has been created for service request

### Any issues with the current functionality (optional):
- had to change the date input component to a select component
- need to add validation
- need to add hourly rate and rating to profile model
- need to finish work on the integration of availability page (#84) since this PR is dependent on it
